scout/server/blueprints/variants/controllers.py

```diff
@@ -434,20 +436,23 @@ def variant_export_genes_info(store, gene_list):
         gene_ids.append(hgnc_id)
         gene_names.append(gene_name)
 
+        if gene_obj.get("canonical_transcript"):
+            canonical_txs.append(gene_obj.get("canonical_transcript"))
+
         hgvs_nucleotide = "-"
+        protein_change = "-"
         # gather HGVS info from gene transcripts
+
         transcripts_list = gene_obj.get("transcripts")
         for transcript_obj in transcripts_list:
-            if (
-                transcript_obj.get("is_canonical") is not None
-                and transcript_obj.get("is_canonical") is True
-            ):
-                hgvs_nucleotide = str(transcript_obj.get("coding_sequence_name"))
+            if transcript_obj["transcript_id"] == gene_obj.get("canonical_transcript"):
```

Comment:
`<rant>`
We should take a couple of h and go over all of these at some point.
- `canonical_transcript` is annotated by VEP, and corresponds to the Canonical transcript status in ENSEMBL: typically the longes transcript around. In many cases this overlaps with, but can absolutely be distinct from primary_transcript - is also typically expressed as ENSEBL transcript.
- `primary_transcript` is the HGNC gene preferred transcript, which should be used for clinical reporting nomenclature unless there are pressing exceptions. Is expressed as a REFSEQ transcript. Now VEP "HGVS" are afaik based on ENSEMBL anyway. For hg38 MANE select unifies RefSeq and ENSEMBL but for 37 we are left with what we have for mapping RefSeq to ENSEML.

In general, when we export only one thing, I think this should actually be `primary_transcript`, and may have to be fetched off the `hgnc_gene` at more points, as it is not currently on the variant-stored gene_obj we use here. 
`</rant>`
In this case, the file very clearly states canonical transcript etc so fine, so I guess it is ok. 😅

Reply:
Yes. For me this is (and will always be) a mess. But I start to grasp the nuances

Reply:
So this is fine to merge or do we want to revisit what should be exported (primary instead of canonical?)

Reply:
I think it is fine. It is pretty clear as is, and was requested mainly from the research side if I recall correctly from the meeting? That makes me feel it's even more ok: it is not as important to stick to HGNC there. For most genes they will be the same anyway, and for a subset of the remainder it will be wrong even with our current implementation of primary. Go forward! We can fix later if called for, and it will make better more sense to fix it once we have 38 for all.